cluster-autoscaler/core/scale_down.go

```diff
@@ -382,9 +481,11 @@ func (sd *ScaleDown) TryToScaleDown(allNodes []*apiv1.Node, pods []*apiv1.Pod, p
 			errors.CloudProviderError,
 			errCP)
 	}
-	coresTotal, memoryTotal := calculateCoresAndMemoryTotal(nodesWithoutMaster, currentTime)
-	coresLeft := coresTotal - resourceLimiter.GetMin(cloudprovider.ResourceNameCores)
-	memoryLeft := memoryTotal - resourceLimiter.GetMin(cloudprovider.ResourceNameMemory)
+
+	scaleDownResourcesLeft, err := computeScaleDownResourcesLeftLimits(nodesWithoutMaster, resourceLimiter, currentTime)
+	if err != nil {
+		return ScaleDownError, err.AddPrefix("failed to obtain nodes resources: ")
```

Comment:
I'd like the error message to be more descriptive. We probably get node resources all over the place and just looking at this error I wouldn't guess it means 'we failed to calculate resource limits for scale-down'.

Reply:
Actually this call should not return error.
We will only mark resources for which we could not compute limit due to error as `limitUnknown`.
Changed.